tools/trace2syz/trace2syz/util.go

```diff
@@ -0,0 +1,857 @@
+		"":                       true,
+		"execve":                 true, // unsupported
+		"access":                 true, // unsupported
+		"gettimeofday":           true, // unsupported
+		"kill":                   true, // unsupported
+		"arch_prctl":             true,
+		"getcwd":                 true, // unsupported
+		"setdomainname":          true, // unsupported
+		"reboot":                 true, // unsupported
+		"getppid":                true, // unsupported
+		"umask":                  true, // unsupported
+		"adjtimex":               true, // unsupported
+		"nanosleep":              true,
+		"wait4":                  true,
+		"wait":                   true,
+		"futex":                  true,
+		"rt_sigprocmask":         true,
+		"rt_sigaction":           true,
+		"rt_sigtimedwait":        true,
+		"rt_sigreturn":           true,
+		"rt_sigqueueinfo":        true,
+		"rt_sigsuspend":          true,
+		"tgkill":                 true,
+		"sysfs":                  true, // unsupported
+		"clone":                  true, // unsupported
+		"newfstatat":             true, // unsupported
+		"getsid":                 true,
+		"getcpu":                 true,
+		"mmap":                   true,
+		"msync":                  true,
+		"mremap":                 true,
+		"mprotect":               true,
+		"madvise":                true,
+		"sched_get_priority_min": true,
+		"sched_get_priority_max": true,
+	}
+
+	acceptLabels = map[string]string{
+		"fd":            "", // TODO: this is an illegal value. how do we interpret the uniontype?
+		"sock":          "",
+		"sock_alg":      "$alg",
+		"sock_in":       "$inet",
+		"sock_in6":      "$inet6",
+		"sock_netrom":   "$netrom",
+		"sock_nfc_llcp": "$nfc_llcp",
+		"sock_sctp":     "$sctp",
+		"sock_unix":     "$unix",
+		"sock_netlink":  "$inet6",
+	}
+
+	bindLabels = map[string]string{
+		"fd":             "",
+		"sock":           "",
+		"sock_alg":       "$alg",
+		"sock_bt_hci":    "$bt_hci",
+		"sock_bt_l2cap":  "$bt_l2cap",
+		"sock_bt_rfcomm": "$bt_rfcomm",
+		"sock_bt_sco":    "$bt_sco",
+		"sock_in":        "$inet",
+		"sock_in6":       "$inet6",
+		"sock_netlink":   "$netlink",
+		"sock_netrom":    "$netrom",
+		"sock_nfc_llcp":  "$nfc_llcp",
+		"sock_sctp":      "$sctp",
+		"sock_unix":      "$unix",
+		"AF_INET":        "$inet",
+		"AF_INET6":       "$inet6",
+		"sock_packet":    "$packet",
+	}
+
+	connectLabels = map[string]string{
+		"fd":             "",
+		"sock":           "",
+		"sock_bt_l2cap":  "$bt_l2cap",
+		"sock_bt_rfcomm": "$bt_rfcomm",
+		"sock_bt_sco":    "$bt_sco",
+		"sock_in":        "$inet",
+		"sock_in6":       "$inet6",
+		"sock_netlink":   "$netlink",
+		"sock_netrom":    "$netrom",
+		"sock_nfc_llcp":  "$nfc_llcp",
+		"sock_nfc_raw":   "$nfc_raw",
+		"sock_sctp":      "$sctp",
+		"sock_unix":      "$unix",
+		"AF_INET":        "$inet",
+		"AF_INET6":       "$inet6",
+	}
+
+	bpfLabels = map[string]string{
+		"BPF_PROG_LOAD":        "$PROG_LOAD",
```

Comment:
Yeah I agree. So is it reasonable to hand select the system calls which need the mappings like 1) socket, 2) ioctl, 3) setsockopt, 4) getsockopt, .... but then generate these mappings on startup? I want to select the most descriptive system call that Syzkaller offers.

Reply:
Hardcoding a small set of syscalls looks more reasonable, the syscall names are stable and will never change.
It's also possible to find all syscalls with discriminations automatically. But it will also give us, say, write syscall. It's also possible to check if a group of discriminations have an int argument that is const in most discriminations. Though, that will filter out e.g. accept.
We could also rename some syscalls for more consistency, e.g. bpf$PROG_LOAD to $BPF_PROG_LOAD so that const actually matches name.